tools/fix_bss.py

```diff
@@ -501,32 +501,103 @@ def solve_bss_ordering(
     raise FixBssException("Could not find any solutions")
 
 
+# Parses #pragma increment_block_number (with line continuations already removed)
+def parse_pragma(pragma_string: str) -> dict[str, int]:
+    amounts = {}
+    for part in pragma_string.replace('"', "").split()[2:]:
+        kv = part.split(":")
+        if len(kv) != 2:
+            raise FixBssException(
+                "#pragma increment_block_number"
+                f' arguments must be version:amount pairs, not "{part}"'
+            )
+        try:
+            amount = int(kv[1])
+        except ValueError:
+            raise FixBssException(
+                "#pragma increment_block_number"
+                f' amount must be an integer, not "{kv[1]}" (in "{part}")'
+            )
+        amounts[kv[0]] = amount
+    return amounts
+
+
+# Formats #pragma increment_block_number as a list of lines
+def format_pragma(amounts: dict[str, int], max_line_length: int) -> list[str]:
+    lines = []
+    pragma_start = "#pragma increment_block_number "
+    current_line = pragma_start + '"'
+    first = True
+    for version, amount in sorted(amounts.items()):
+        part = f"{version}:{amount}"
+        if len(current_line) + len(part) + len('" \\') > max_line_length:
+            lines.append(current_line + '" \\\n')
+            current_line = " " * len(pragma_start) + '"'
+            first = True
+        if not first:
+            current_line += " "
+        current_line += part
+        first = False
+    lines.append(current_line + '"\n')
+
+    if len(lines) >= 2:
+        # align all continuation \ characters vertically
```

Comment:
could this be simplified by not adding the `\` in the loop above and adding them here instead?

Reply:
Indeed

Reply:
Sorry, move the space too, so that `line[:-2]` can just be `line`?